### ISSUE TITLE
Not working after spotcast update

### ISSUE THREAD

**Issue by SkiBoard** [custom-cards/spotify-card#215]
Yesterday I updated spotcast (https://github.com/fondberg/spotcast) to the latest version 3.8.2 and now the Spotify-card just shows Loading... in the spot where the grid of albums once sat, and when I select "choose player" it says "no devices found" 

* [x] Reproducable on a plain lovelace panel without card, style or theme mods
* [x] Using latest version of spotcast and spotify-card [x]
* [x] Using latest stable version of Home Assistant
* [x] I have Spotify Premium
* [x] I have setup the Spotify integration in Home Assistant

**To Reproduce**
Install the card, integrations and add-ins as directed in spotcast and spotify-card (note, I didn't get up to doing the automations noted here (https://community.home-assistant.io/t/using-your-spotify-account-in-a-ha-dashboard-to-select-speakers-and-broadcast/487300) as I was not able to see the speaker list here: 
![image](https://github.com/user-attachments/assets/8023a911-e518-4a34-b3ff-257e6dc3838a)

**Expected behavior**
When I load the dashboard card I expect to see something like this: 
![image](https://github.com/user-attachments/assets/e9ec9aba-1d28-48c9-9e5f-6156ff84f606)

where I can click the speaker and see all the speakers (I was able to do that before the update). 

Instead I see this: 
![image](https://github.com/user-attachments/assets/c5aa9930-5cc2-4678-905b-1bc11d999b87)

Note that it's showing the playing track but the grid is not showing and the speaker selector says "no devices found" (note that the standard spotify card shows two DLNA devices but not the chrome cast devices or groups - before the update I saw all speakers, devices and groups.  

**Environment (please complete the following information):**
 - Windows
 - Browser: Chrome
 - Version: [ ]
 - HA version: Core 2024.9.3
 - spotcast version: 3.8.2

**Comment by Electronlibre2012** [comment]
seems we are alone until @fondberg has time to respond and fix it...so bad, there is no other great card for spotify like his one...

**Comment by joq3** [comment]
Same issue here :(

**Comment by SkiBoard** [comment]
That’s a shame. Though it’s somewhat comforting knowing it’s not just me. Hopefully we’ll see a fix for it soon.

**Comment by bmx170** [comment]
commenting to get notification of when this will be fixed.

**Comment by geoffoxholm** [comment]
> commenting to get notification of when this will be fixed.

Future folks can just click "Subscribe" at the bottom of the right-column.

**Comment by PerSterin** [comment]
**Maintainer passed away.** 

https://github.com/fondberg/spotcast/issues/425 

My condolences to @fondberg friends and family.

**Comment by SkiBoard** [comment]
I’m also very sorry to hear that. My condolences.  

Do we know if anyone has taken it over? 

If not, is anyone aware of a fix? I assume I can just roll back to a previous version of Spotcast?

**Comment by Electronlibre2012** [comment]
so sad, my condoléances to @fondberg and his family

**Comment by PerSterin** [comment]
If you have this error, change line 146 of __init__.py. and restart HA. See error in logger below.

From: resp = get_spotify_devices(spotify_media_player)
To: resp = get_spotify_devices(spotify_media_player, hass)

This solved my problem

Logger: homeassistant
Source: custom_components/spotcast/__init__.py:146
integration: Spotcast (documentation, issues)
First occurred: 15:38:48 (4 occurrences)
Last logged: 15:39:16

Error doing job: Task exception was never retrieved (None)
Traceback (most recent call last):
  File "/config/custom_components/spotcast/helpers.py", line 122, in run
    return await loop.run_in_executor(executor, pfunc)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/spotcast/__init__.py", line 146, in get_devices
    resp = get_spotify_devices(spotify_media_player)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: get_spotify_devices() missing 1 required positional argument: 'hass'

**Comment by Electronlibre2012** [comment]
thanks @PerSterin ! that work as before

**Comment by PerSterin** [comment]
@geoffoxholm
Opened PR https://github.com/fondberg/spotcast/pull/462 to address this.

Thanks
Great then we will get a new Spotcast release soon.

**Comment by IIIdefconIII** [comment]
would be loevely to have this working again, my condulances!

**Comment by krzakx** [comment]
@PerSterin works for me too! Thanks.

**Comment by SkiBoard** [comment]
> If you have this error, change line 146 of **init**.py. and restart HA. See error in logger below.
> 
> From: resp = get_spotify_devices(spotify_media_player) To: resp = get_spotify_devices(spotify_media_player, hass)
> 
> This solved my problem
> 
> Logger: homeassistant Source: custom_components/spotcast/**init**.py:146 integration: Spotcast (documentation, issues) First occurred: 15:38:48 (4 occurrences) Last logged: 15:39:16
> 
> Error doing job: Task exception was never retrieved (None) Traceback (most recent call last): File "/config/custom_components/spotcast/helpers.py", line 122, in run return await loop.run_in_executor(executor, pfunc) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run result = self.fn(*self.args, **self.kwargs) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/spotcast/**init**.py", line 146, in get_devices resp = get_spotify_devices(spotify_media_player) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ TypeError: get_spotify_devices() missing 1 required positional argument: 'hass'

Worked for me too!   Thanks so much.   I made a note of this in a forked Readme on the Spotify-Card Git

**Comment by alexschultze** [comment]
Thank you this fix with line 146 works.

I am a new user and it took my several hours to find the bug ! Please bring this upstream asap.

**Comment by PerSterin** [comment]
@geoffoxholm Opened PR https://github.com/fondberg/spotcast/pull/462 to address this.
It is awaiting for approval / merging to the main branch from the maintainer. @fcusson

**Comment by fcusson** [comment]
@SkiBoard can you confirm if the update fixes the issue for you?

**Comment by SkiBoard** [comment]
Yes that worked thanks so much!On 31 Oct 2024, at 05:53, Felix Cusson ***@***.***> wrote:﻿
@SkiBoard can you confirm if the update fixes the issue for you?

**Comment by fcusson** [comment]
@SkiBoard can you close the ticket?

**Comment by SkiBoard** [comment]
Closed, thank you. Apologies for that. I’m new to git hub On 31 Oct 2024, at 22:44, Felix Cusson ***@***.***> wrote:﻿
@SkiBoard can you close the ticket?

**Comment by fcusson** [comment]
@SkiBoard no stress, if I had maintainer access I would have done it but I don't think Fondberg gave access to anyone, so we can't really close these tickets ourselves.

I just didn't want the ticket to mislead some into believing this specific issue was still live

**Comment by SkiBoard** [comment]
Ok - so I just updated to 4.0.0 and the same issue returned.  I tried finding the in init.py "From: resp = get_spotify_devices(spotify_media_player)" but it was already set to "To: resp = get_spotify_devices(spotify_media_player, hass) ".

I rolled back to 3.9.0 and that fixed it.  Any suggestions to make 4.0.0 work?

**Comment by fcusson** [comment]
@SkiBoard. The change in Spotify Library for the main integration broke spotify-card, this isn't something that can be fixed through Spotcast anymore. Also, `3.9.0` will not work at all in `2024.11` so I don,t understand what *works*.

Anyway @mikevanes created a fork ok the project with a fix [here](https://github.com/mikevanes/spotify-card-v2).

**Comment by Electronlibre2012** [comment]
@fcusson i think @SkiBoard means with reverted to 3.9.0 spotify-card works : show playlists and  all my media-players, but when i select a media-players, i heard the beep of the google home i have select and after it doesnt start to play and show this error : Échec de l'exécution de l'action spotcast/start. 'Device' object is not subscriptable.

with v 4.0, the card doesnt work but when going in the media-player directly and browse media the music doesnt start and show non error message...

**Comment by mikevanes** [comment]
As @fcusson already mentioned, my fork works with the latest spot cast. Also adds a new playlist feature "recently-played" @Electronlibre2012 @SkiBoard

**Comment by mikevanes** [comment]
@fcusson could you add me on discord perhaps so that when the time comes for v5 we can easily keep in touch to make sure nothing breaks? mikeve97

**Comment by fcusson** [comment]
@SkiBoard @Electronlibre2012, then that would be a spotcast issue not a Spotify-card one, please open a ticket in spotcast with all the relevant information.

With the error you are mentioning it looks like you have a mismatch of versions.

It is very important that you have only have `v4.0.0`installed with `2024.11`

The device error would happen if you have the `v3.9.0` code but with Spotify marked as a dependency, which will not work by itself

**Comment by Electronlibre2012** [comment]
> As @fcusson already mentioned, my fork works with the latest spot cast. Also adds a new playlist feature "recently-played" @Electronlibre2012 @SkiBoard

Sorry @mikevanes , i thought it doesnt work, it work...but it takes up to 5 seconds or more to begin cast music on google home.

**Comment by Electronlibre2012** [comment]
> With the error you are mentioning it looks like you have a mismatch of versions.
> 
> It is very important that you have only have `v4.0.0`installed with `2024.11`
> 
> The device error would happen if you have the `v3.9.0` code but with Spotify marked as a dependency, which will not work by itself

You are right, i have just try to revert to 3.9.0 to see how it works. The spotify-card is so nice...dont have to go in the media-player, media-files, then spotify, then playlist and choose a playlist. With the spotify-card, just select the media-player in the list, then playlist, only 2 clicks :)

its not a big deal...most important is that it works and play play music from spotify, thanks a lot @fcusson and @mikevanes for your work!

**Comment by SkiBoard** [comment]
@fcusson @mikevanes Thank you so much.  I've updated to Spotcast 4.0.0 and now loaded your fork which works perfectly.  I'll stick to this version going forward. Thanks again for all your work!

@mikevanes  If I may make one small suggestion for your amazing card?  Opening the list of devices in the "choose player" on the card might be better on click (rather than on mouseover) as when I use a touch screen if I just "tap" that option, it selects the closest player to where I tapped (unless I tap and drag my finger).  Tapping to open the list then scrolling / clicking the desired player feels more intuitive.  Perhaps give the user an option to select on hover or on click?  Not a biggie - as I said, I LOVE your card.  Thanks again.  I'll close my ticket for now.
